x-pack/plugins/infra/server/routes/available_fields/helpers/get_sample_field_names.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AvailableFieldsAggregation, AvailableFieldsBucket, AvailableFieldsHit } from '../types';
+import { extractFields } from './extract_fields';
+import { InfraDatabaseSearchResponse } from '../../../lib/adapters/framework';
+
+export const getSampleFieldNames = async (
+  search: <Aggregation>(options: object) => Promise<InfraDatabaseSearchResponse<{}, Aggregation>>,
+  options: { timeField: string; indexPattern: string }
+): Promise<string[]> => {
+  const params = {
+    index: options.indexPattern,
+    body: {
+      query: {
+        range: {
+          [options.timeField]: {
+            gte: 'now-5m',
```

Comment:
What happens if we would pull in the data for the currently selected range instead? Would this become too slow for larger periods?

Reply:
I don't think this will become too slow. After some thought last night I was planning on adding the time range to the query.